### PR TITLE
Clean up some code in node validators

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

Clean up some code in node validators

### CODE REVIEW

**Comment by robobun** [comment]
@Jarred-Sumner, your commit 7bd837dc057eb3ee9672728a116ec7bf711c7334 is building: [`#5418`](https://buildkite.com/bun/bun/builds/5418)

**Review by nektro — APPROVED** [review]

File: src/bun.js/bindings/NodeValidator.cpp [hunk 1]
```diff
@@ -378,19 +391,32 @@ JSC_DEFINE_HOST_FUNCTION(jsFunction_validateEncoding, (JSC::JSGlobalObject * glo
...
+    if (normalized == BufferEncodingType::hex) {
+        auto data = callFrame->argument(0);
+
+        size_t length = 0;
+        if (data.isString()) {
+            length = data.toString(globalObject)->length();
+        } else if (auto* view = JSC::jsDynamicCast<JSC::JSArrayBufferView*>(data)) {
+            length = view->length();
+        } else if (auto* buffer = JSC::jsDynamicCast<JSC::JSArrayBuffer*>(data)) {
+            if (auto* impl = buffer->impl()) {
```
Comment (nektro) [inline src/bun.js/bindings/NodeValidator.cpp:406]: when would the first condition be true but this one be false?

Comment (Jarred-Sumner) [reply src/bun.js/bindings/NodeValidator.cpp:406]: `ArrayBuffer` != `ArrayBufferView`, `Uint8Array` would fit `ArrayBufferView` but not `ArrayBuffer`.

Comment (nektro) [reply src/bun.js/bindings/NodeValidator.cpp:406]: yes, i meant specifically `if (auto* buffer =` and `if (auto* impl =`

**Pull request merged**